src/irr/asset/COBJMeshFileLoader.cpp

```diff
@@ -355,490 +734,377 @@ asset::SAssetBundle COBJMeshFileLoader::loadAsset(io::IReadFile* _file, const as
 		// eat up rest of line
 		bufPtr = goNextLine(bufPtr, bufEnd);
 	}	// end while(bufPtr && (bufPtr-buf<filesize))
-	// Clean up the allocate obj _file contents
-	delete [] buf;
-
-	asset::CCPUMesh* mesh = new asset::CCPUMesh();
 
-	// Combine all the groups (meshbuffers) into the mesh
-	for ( uint32_t m = 0; m < ctx.Materials.size(); ++m )
-	{
-        {//arbitrary scope
-            auto preloadedMbItr = ctx.preloadedSubmeshes.find(ctx.Materials[m]);
-            if (preloadedMbItr != ctx.preloadedSubmeshes.end())
-            {
-                mesh->addMeshBuffer(core::smart_refctd_ptr<ICPUMeshBuffer>(preloadedMbItr->second,core::dont_grab));
-                preloadedMbItr->second->drop(); // after grab when we got it from cache
-                continue;
-            }
-        }
+    core::vector<core::vectorSIMDf> newNormals;
+    auto doRecalcNormals = [&vertices,&newNormals](const core::vector<uint32_t>& _ixs) {
```

Comment:
I wonder if there's any point in computing these rough normals, if we have the geometry manipulator function to smooth meshes.

Reply:
?

Reply:
I repeat the question

Reply:
yea, i figured out how to use it and made use of `calculateSmoothNormals()`

Reply:
so what's this lambda still doing here?

Reply:
sorry, didnt push yesterday